x/programs/runtime/memory.go

```diff
@@ -111,7 +111,7 @@ func (m *memory) Len() (uint64, error) {
 
 // WriteBytes is a helper function that allocates memory and writes the given
 // bytes to the memory returning the offset.
-func WriteBytes(m Memory, buf []byte) (uint64, error) {
+func WriteBytes(m Memory, buf []byte) (int64, error) {
```

Comment:
this is a bit awkward because a memory location _should_ be a uint64. However, it might makes sense to keep the `int64` here to avoid potential overflow errors. We pass these memory pointers  into `wasm` which will convert to `int64`, so if we somehow return a `ptr` here that's greater than `MaxInt64` we will have undefined behavior.

Reply:
I guess I would need to dig into this further but my general understand is that while technically possible for a pointer to exceed that limit. A 64bit system would not utilize even close to that entire range and a Program which has a capped memory limit via the runtime store/engine is a much smaller subnet of that address space.

I guess what I am saying is there is a trade off between extreme edge case safety and usability/complexity. Its a great topic though and we should try to write some tests to exercise this.

Reply:
I can write this up as an issue for a deeper dive later, or should I address/look into this now? I find it a bit unsettling that the host can return pointers our wasm programs wont be able to reference.

Reply:
linked #605